### ISSUE TITLE
Stardew Valley

### ISSUE THREAD

**Issue by Zielakless** [Rinnegatamante/Android2Vita-Candidate-Ports-List#431]
### Submission Rules

- [X] I have read the README.md and understand the submission rules

### No Unity or Java Games

- [X] This game is NOT made in Unity and does not contain a libunity.so
- [X] This game is NOT made in Java and does not contain a libgdx.so

### No Duplicate Submissions

- [X] This game is NOT already in the Android2Vita candidate port list

### Game Name

Stardew Valley

### Game Version

1.5.6.39

### ARM Build Version

ARMv7

### OpenGL Version

GLES 1

### FMOD Usage

No FMOD

### Listed or Delisted

Google Play Store

### Store Link

https://play.google.com/store/apps/details?id=com.chucklefish.stardewvalley&hl=pl&gl=US

**Comment by treasboi** [comment]
We already have Stardew Valley for psvita

**Comment by CpazR** [comment]
> We already have Stardew Valley for psvita

This is true. But I do think the mobile version is still being supported through updates?
I know for sure the vita version isn't supported anymore.

**Comment by radimovotajemno** [comment]
It would be awesome to have a port of Android SV on the Vita!

**Comment by MonteXMat** [comment]
That'd be sick.

**Comment by radimovotajemno** [comment]
will someone work on it?

**Comment by wombajanga** [comment]
+1 would be awesome

**Comment by Primitheus** [comment]
This idea peaked my interest due to the PS Vita's Official Port being outdated and buggy. Furthermore, if this were to be done it could bring potential for a SMAPI port too and enable full Modding support. So I decided to do a bit of research. 

Porting this game is a bigger task than (at least I) expected, due to the use of Mono/MonoGame and SkiaSharp Graphics API (with I believe GLESv2 as the backend, not GLESv1 as I think the AndroidManifest.xml is incorrect). 

There exists a WIP for a Mono wrapper called VitaMonoLoader but from what I understand, there's still a significant amount of work that would need to be done for a Stardew Valley Port.

But because the game uses .NET, there is the other option which would be to Decompile and Recompile the game for Vita. 

Any how, feel free to contact me if you think or would like to help me out or correct me on anything, and in the unlikely event I make significant progress in the port I will update this and direct to the project.

**Comment by nekoteai** [comment]
@Primitheus good investigation! I think monoloader maybe a faster way. As Android port .NET may slow down effeiciently and hard for debug. I found a wonderful port using mono runtime on handheld Linux console,which is able to run. https://portmaster.games/detail.html?name=stardewvalley
They Utilized a Loader and port GLes libs.
Seems if we have monoloader which support Vita resources correctly,then it can be run.

**Comment by Primitheus** [comment]
@nekoteai I've been unable to utilize the WIP [VitaMonoLoader](https://github.com/GrapheneCt/VitaMonoLoader/tree/main), I can't seem to get it setup correctly to use in my projects. Even then I do not know how efficient or complete the VitaMonoLoader is, never the less I would like to investigate the repo closer.

The other option for a Mono Loader for Vita would be PSM Studio's Loader but that would also require extensive work to decompile and recompile and as a whole not incredibly viable.

Seems like it's still a long way to go for this port project and a little hopeless as I am not nearly knowledgeable enough to make any significant process. Never the less I am still interested in learning more and hopefully get something up and running eventually (fingers crossed).

**Comment by HyronY** [comment]
i would like to add that android version 1.6.x is not possible to port.
(no lib folders)

**Comment by Primitheus** [comment]
> i would like to add that android version 1.6.x is not possible to port. (no lib folders)

That is not true, the 1.6.x and all previous versions contain the lib folder. If you're referring to the ".dlls", they are also included in full within the APK. They're just packed into the /unknown/assemblies. The dlls are completely extractable with no issues. Both the Android and PC versions of the game can be ported to Vita.

**Comment by HyronY** [comment]
> > i would like to add that android version 1.6.x is not possible to port. (no lib folders)
> 
> That is not true, the 1.6.x and all previous versions contain the lib folder. If you're referring to the ".dlls", they are also included in full within the APK. They're just packed into the /unknown/assemblies. The dlls are completely extractable with no issues. Both the Android and PC versions of the game can be ported to Vita.

yes you are corrrect (YAY!), my (big) mistake.  Good to know both versions (android & pc) are still possible for vita

![2024-11-21 08_32_46-assemblies - Verkenner](https://github.com/user-attachments/assets/1576b824-4719-4093-96b8-07c731751c0f)

**Comment by kayzidev** [comment]
Hey mates ! So what's the statue of the port few months later ? Is this still possible ? Thanks for your work btw

**Comment by SynToWin0522** [comment]
> Hey mates ! So what's the statue of the port few months later ? Is this still possible ? Thanks for your work btw

I came to ask the same thing, Stardew Valley Vita has been a portable joy to play and as im entering Summer of year 2, im slowly realizing that features and especially items (looking at you, triple shot espresso) I enjoy aren't in the game.

**Comment by Elthisboy** [comment]
Hiii!
Just wondering if there’s been any update on this candidate port for Stardew Valley? It seems like a really promising game to bring to Vita given its ARMv7/GLES1 support and no FMOD issues  and it’s been quite a while with no replies here

Also curious! would it be possible to aim for online play or even mod support if this gets ported? Having those features on Vita could add a lot of replay value and community interest, especially for a game like Stardew where multiplayer and mods are big parts of the experience on other platforms.

Thanks in advance for any insight
